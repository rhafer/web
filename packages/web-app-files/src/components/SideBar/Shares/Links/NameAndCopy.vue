<template>
  <div class="oc-mb-s oc-width-1-1">
    <h4 class="oc-text-truncate oc-files-file-link-name oc-my-s" v-text="linkName" />
    <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1 oc-p-s link-name-container">
      <div class="oc-flex oc-flex-middle oc-text-truncate">
        <oc-icon name="link" fill-type="line" />
        <p class="oc-files-file-link-url oc-ml-s oc-text-truncate oc-my-rm" v-text="link.url" />
      </div>
      <copy-to-clipboard-button
        class="oc-files-public-link-copy-url"
        :value="link.url"
        :label="copyToClipboardLabel"
        :success-msg-title="copyToClipboardSuccessMsgTitle"
        :success-msg-text="getCopyToClipboardSuccessMsgText(linkName)"
      />
    </div>
  </div>
</template>

<script>
import CopyToClipboardButton from '../../Shared/CopyToClipboardButton.vue'

export default {
  name: 'NameAndCopy',
  components: { CopyToClipboardButton },
  props: {
    link: {
      type: Object,
      required: true
    }
  },
  computed: {
    linkName() {
      return this.link.name
    },

    copyToClipboardLabel() {
      return this.$gettext('Copy link to clipboard')
    },

    copyToClipboardSuccessMsgTitle() {
      return this.$gettext('Link copied')
    }
  },
  methods: {
    getCopyToClipboardSuccessMsgText(linkName) {
      return this.$gettextInterpolate(
        this.$gettext('The link "%{linkName}" has been copied to your clipboard.'),
        {
          linkName
        },
        true
      )
    }
  }
}
</script>

<style scoped>
.link-name-container {
  background-color: var(--oc-color-input-bg);
  border: 1px solid var(--oc-color-input-border);
}
</style>
