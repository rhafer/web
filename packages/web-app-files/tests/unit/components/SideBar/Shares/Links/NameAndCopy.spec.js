import { createLocalVue, shallowMount } from '@vue/test-utils'
import GetTextPlugin from 'vue-gettext'
import DesignSystem from 'owncloud-design-system'
import NameAndCopy from '@files/src/components/SideBar/Shares/Links/NameAndCopy.vue'

const localVue = createLocalVue()
localVue.use(DesignSystem)
localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

const exampleLink = {
  name: 'Example link',
  url: 'https://some-url.com/abc'
}

describe('NameAndCopy', () => {
  it('should show link info component', () => {
    const wrapper = getShallowWrapper()
    expect(wrapper).toMatchSnapshot()
  })
})

function getShallowWrapper() {
  return shallowMount(NameAndCopy, {
    localVue,
    propsData: {
      link: exampleLink
    },
    stubs: {
      'oc-icon': true,
      'copy-to-clipboard-button': true
    }
  })
}
