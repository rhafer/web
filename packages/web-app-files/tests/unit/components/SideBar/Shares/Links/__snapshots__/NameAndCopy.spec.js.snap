// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameAndCopy should show link info component 1`] = `
<div class="oc-mb-s oc-width-1-1">
  <h4 class="oc-text-truncate oc-files-file-link-name oc-my-s">Example link</h4>
  <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1 oc-p-s link-name-container">
    <div class="oc-flex oc-flex-middle oc-text-truncate">
      <oc-icon-stub name="link" filltype="line" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub>
      <p class="oc-files-file-link-url oc-ml-s oc-text-truncate oc-my-rm">https://some-url.com/abc</p>
    </div>
    <copy-to-clipboard-button-stub text="" value="https://some-url.com/abc" label="Copy link to clipboard" successmsgtitle="Link copied" successmsgtext="The link &quot;Example link&quot; has been copied to your clipboard." class="oc-files-public-link-copy-url"></copy-to-clipboard-button-stub>
  </div>
</div>
`;
